README.md

```diff
@@ -59,136 +69,70 @@ According to original BEM-naming convention it looks like the following:
-}
+'block__some-elem__sub-elem'
 ```
 
-API
----
+For more information, see the [FAQ](https://en.bem.info/methodology/faq/):
 
-* [parse(str)](#parsestr)
-* [stringify(obj)](#stringifyobj)
-* [elemDelim](#elemdelim)
-* [modDelim](#moddelim)
-* [modValDelim](#modvaldelim)
-
-### parse(str)
+> [Why does BEM not recommend using elements within elements (block__elem1__elem2)?](https://en.bem.info/methodology/faq/#why-does-bem-not-recommend-using-elements-within-elements-block__elem1__elem2)
 
-It parses string into naming object.
-
-Example:
+Also there is no such BEM entity as a modifier and an element modifier simultaneously so the following string representation will be invalid:
 
 ```js
-const bemNaming = require('bem-naming');
-
-bemNaming.parse('block__elem_mod_val');  // { block: 'block', elem: 'elem',
-                                         //   modName: 'mod', modVal: 'val' }
+'block_block-mod-name_block-mod-val__elem-name_elem-mod-name_elem-mod-val'
 ```
 
-### stringify(obj)
+Convention by Harry Roberts
+---------------------------
+
+According to this convention elements are delimited with two underscores (`__`), modifiers are delimited by two hyphens (`--`), and values of modifiers are delimited by one underscore (`_`).
 
-It forms a string according to naming object.
+Read more in the [Guidelines](http://cssguidelin.es/#bem-like-naming).
 
 Example:
 
 ```js
-const bemNaming = require('bem-naming');
-
-bemNaming.stringify({
-    block: 'block', elem: 'elem',
-    modName: 'mod', modVal: 'val'
-}); // 'block__elem_mod_val'
-```
-
-### elemDelim
-
-String to separate elem from block.
+const twoDashesNaming = require('bem-naming')('two-dashes');
 
-### modDelim
-
-String to separate modifiers from blocks and elements.
-
-### modValDelim
-
-String to separate value of modifier from name of modifier.
-
-Common misconceptions
----------------------
-
-BEM methodology involves the use of flat structure inside a block. It means that BEM entity can not be represented as an element of the other element and the following string representation will be invalid:
-
-```js
-'block__some-elem__sub-elem'
-```
+twoDashesNaming.parse('block__elem');    // { block: 'block', elem: 'elem' }
+twoDashesNaming.parse('block--mod_val'); // { block: 'block',
+                                         //   mod: { name: 'mod', val: 'val' } }
 
-Also there is no such BEM entity as a modifier and an element modifier simultaneously so the following string representation will be invalid:
+twoDashesNaming.stringify({
+    block: 'block',
+    elem: 'elem',
+    mod: 'mod'
+});
 
-```js
-'block_block-mod-name_block-mod-val__elem-name_elem-mod-name_elem-mod-val'
+// ➜ block__elem--mod
 ```
 
 Custom naming convention
 ------------------------
 
-Use `bemNaming` function to create instance to manage naming of your own naming convention.
+Use [bemNaming](#bemnaming-elem-mod-wordpattern-) function to create instance to manage naming of your own naming convention.
```

Comment:
Use the [bemNaming](#bemnaming-elem-mod-wordpattern-) function to create an instance where you can manage your own naming convention.